docarray/document/mixins/property.py

```diff
@@ -87,6 +89,16 @@ def matches(self, value: 'DocumentArray'):
 
         self._data.matches = value
 
+    @_PropertyMixin.embedding.setter
+    def embedding(self, value: 'ArrayType'):
+
+        if is_vector(value):
```

Comment:
this can be a little overkill for every document.

Reply:
Yes I agree